ansible/roles/nginx/templates/nginx.conf.j2

```diff
@@ -16,6 +16,15 @@ http {
         '$http_referer $http_user_agent $upstream_addr';
     access_log /logs/nginx_access.log combined-upstream;
 
+    upstream controllers {
+        server {{ groups['controllers'] | first }}:{{ controller.port }};
+{% for ip in groups['controllers'] %}
+    {% if groups['controllers'].index(ip) > 0 %}
+        server {{ ip }}:{{ controller.port + groups['controllers'].index(ip) }} backup;
```

Comment:
Can we be sure the backup is only used when primary crashed? Failure for nginx is just failure to establish a connection upstream which doesn't necessarily say primary is down - just unresponsive. So it's possible the caches will slip and eventually have to become consistent again.